working_with_the_llvm_ast.md

```diff
@@ -0,0 +1,215 @@
+## Working With the LLVM AST and Using it to Make Changes to Clang-Tidy Source
```

Comment:
Should all this content be added to or at the very least, linked into the README.md so it is easy to find for us?